pkg/controller/configstatus/configstatus_controller.go

```diff
@@ -0,0 +1,195 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configstatus
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/go-logr/logr"
+	configv1alpha1 "github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/status/v1beta1"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/logging"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/util"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/watch"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller").WithValues(logging.Process, "config_status_controller")
+
+type Adder struct {
+	WatchManager *watch.Manager
+}
+
+func (a *Adder) InjectControllerSwitch(_ *watch.ControllerSwitch) {}
+
+func (a *Adder) InjectTracker(_ *readiness.Tracker) {}
+
+// Add creates a new config Status Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (a *Adder) Add(mgr manager.Manager) error {
```

Comment:
It doesn't look like we're gating status controller on whether the `status` operation is enabled -- this is bad because it means this controller will not run as a singleton, which invites write conflicts particularly as the # of pods scales.

Example of the status gate:

https://github.com/open-policy-agent/gatekeeper/blob/07127f2352e9793a8471bfd9d1e15e97d69bf81b/pkg/controller/constrainttemplate/constrainttemplate_controller.go#L155-L178

Though a more appropriate code shape for this PR is probably the mutator status gate:

https://github.com/open-policy-agent/gatekeeper/blob/07127f2352e9793a8471bfd9d1e15e97d69bf81b/pkg/controller/mutatorstatus/mutatorstatus_controller.go#L60-L62

(however we should depend on `Status`, not `MutatorStatus`)

We should also fix this for the expansion template status controller (worth verifying it has a similar oversight first).

Long-term a more uniform design pattern for adding status controllers may help avoid similar oversights in the future, but that's beyond the scope of this PR.

Reply:
Updated the logic to use status gate

Reply:
Can we create PRs to add it to the other status controllers that do not have a similar gate? This would include expansion template status at the minimum.